src/audio/mixin_mixout/mixin_mixout.c

```diff
@@ -182,46 +182,17 @@ static int mix_and_remap(struct comp_dev *dev, const struct mixin_data *mixin_da
 
 	sink_config = &mixin_data->sink_config[sink_index];
 
-	if (sink_config->mixer_mode == IPC4_MIXER_NORMAL_MODE) {
-		/* Mix streams. mix_channel() is reused here to mix streams, not individual
-		 * channels. To do so, (multichannel) stream is treated as single channel:
-		 * channel count is passed as 1, channel index is 0, frame indices (start_frame
-		 * and mixed_frame) and frame count are multiplied by real stream channel count.
-		 */
-		mixin_data->normal_mix_channel(sink, start_frame * audio_stream_get_channels(sink),
-					       mixed_frames * audio_stream_get_channels(sink),
-					       source,
-					       frame_count * audio_stream_get_channels(sink),
-					       sink_config->gain);
-	} else if (sink_config->mixer_mode == IPC4_MIXER_CHANNEL_REMAPPING_MODE) {
-		int i;
-
-		for (i = 0; i < audio_stream_get_channels(sink); i++) {
-			uint8_t source_channel =
-				(sink_config->output_channel_map >> (i * 4)) & 0xf;
-
-			if (source_channel == 0xf) {
-				mixin_data->mute_channel(sink, i, start_frame, mixed_frames,
-							 frame_count);
-			} else {
-				if (source_channel >= audio_stream_get_channels(source)) {
-					comp_err(dev, "Out of range chmap: 0x%x, src channels: %u",
-						 sink_config->output_channel_map,
-						 audio_stream_get_channels(source));
-					return -EINVAL;
-				}
-				mixin_data->remap_mix_channel(sink, i,
-							      audio_stream_get_channels(sink),
-							      start_frame, mixed_frames,
-							      source, source_channel,
-							      audio_stream_get_channels(source),
-							      frame_count, sink_config->gain);
-			}
-		}
-	} else {
-		comp_err(dev, "Unexpected mixer mode: %d", sink_config->mixer_mode);
-		return -EINVAL;
-	}
+	/* Mix streams. mix_channel() is reused here to mix streams, not individual
+	 * channels. To do so, (multichannel) stream is treated as single channel:
+	 * channel count is passed as 1, channel index is 0, frame indices (start_frame
+	 * and mixed_frame) and frame count are multiplied by real stream channel count.
+	 */
+	mixin_data->normal_mix_channel(sink, start_frame * audio_stream_get_channels(sink),
+				       mixed_frames * audio_stream_get_channels(sink),
+				       source,
+				       frame_count * audio_stream_get_channels(sink),
```

Comment:
I see a problem here introduced earlier not by this PR. The declaration of
normal_mix_channel_s16() in mixin_mixout_generic.c is:

static void normal_mix_channel_s16(struct audio_stream __sparse_cache *sink, int32_t start_frame,
				   int32_t mixed_frames,
				   const struct audio_stream __sparse_cache *source,
				   int32_t frame_count, uint16_t gain)

However, for frame_count parameter frame_count * audio_stream_get_channels(sink) is passed by the caller. So that is a sample count, not frame_count. The code only works properly because normal_mix_channel_s16() treats frame_count as sample_count. Anyway, that is very confusing and all normal_mix_channel_sXX() should be fixed with renaming frame_count with sample_count (same problem with start_frame and mixed_frames parameters which are also treated as start_sample and mixed_samples). In addition, since after some previous modification normal_mix_channel_s16() has nothing to do with channels (it iterates by samples and mixes entire streams) that function could be renamed to just mix_s16().

This is not a problem introduced by this PR and could be addressed in a separate PR.

Reply:
adding @singalsu and @andrula-song for this new remapping problem. Do we need a new issue to track ? If so, lets add.